WebKit/WebKit
# Generate serialization of NSShadow

#### c7ecac1db21bec1e86214988a146ecd119be54b2
<pre>
Generate serialization of NSShadow
<a href="https://bugs.webkit.org/show_bug.cgi?id=268963">https://bugs.webkit.org/show_bug.cgi?id=268963</a>
&lt;<a href="https://rdar.apple.com/107553244">rdar://107553244</a>&gt;

Reviewed by Brady Eidson.

* Source/WebCore/PAL/pal/ios/UIKitSoftLink.mm:
- Export PAL::getNSShadowClass() for use in WebKit.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
- Add CoreIPCNSShadow.serialization.in.
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:
- Add predeclaration for NSShadow class.
- Add NSShadow to enum class NSType.
(IPC::getClass&lt;PlatformColor&gt;): Add.
(IPC::getClass&lt;NSShadow&gt;): Add.
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::getClass&lt;PlatformColor&gt;): Add.
- Provide method to get UIColor.class since it&apos;s soft-linked on
  non-AppKit platforms.
(IPC::getClass&lt;NSShadow&gt;): Add.
- Provide method to get NSShadow.class since it&apos;s soft-linked on
  non-AppKit platforms.
(IPC::typeFromObject):
- Switch from WebCore::CocoaColor to PlatformColorClass.
- Update to handle NSShadow.
(IPC::shouldEnableStrictMode):
- Remove reference to NSShadow.
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.mm:
(WebKit::valueFromID):
- Add CoreIPCNSShadow.
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h: Copied from Source/WebKit/Shared/Cocoa/CoreIPCTypes.h.
(WebKit::CoreIPCNSShadow::CoreIPCNSShadow):
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.mm: Copied from Source/WebKit/Shared/Cocoa/CoreIPCTypes.h.
(WebKit::CoreIPCNSShadow::CoreIPCNSShadow):
(WebKit::CoreIPCNSShadow::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.serialization.in: Added.
* Source/WebKit/Shared/Cocoa/CoreIPCTypes.h:
- Include CoreIPCNSShadow.h.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
- Add CoreIPCNSShadow.* files to the project.

* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(ObjCHolderForTesting::encode const):
- Add RetainPtr&lt;NSShadow&gt; to variant for testing.
(runTestNS):
(runTestCFWithExpectedResult):
(runTestCF):
- Change lambdas to static functions so we could start to break apart
  the IPC.Serialization.Basic test, which is way too big.
(TEST(IPCSerialization, Basic)):
- Move lambdas out of of this test function.
(TEST(IPCSerialization, NSShadow)): Add.
- Test serialization of NSShadow.
(TEST(IPCSerialization, SecureCoding)):
- Remove duplicate runTestNS lambda.

Canonical link: <a href="https://commits.webkit.org/274870@main">https://commits.webkit.org/274870@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/9cf09a2a8178f7873029c8683b07da68cb16f361

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40198 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19210 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42576 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42743 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/36287 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42505 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22133 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16539 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/33414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40772 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16184 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34713 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/13983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/14026 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35665 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44021 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36465 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35994 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39741 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/15013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12314 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/38019 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16632 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/16681 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5322 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/16276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | |
